### PR TITLE
Bring Vale into the future

### PR DESCRIPTION
**Pull request by nickvigilante**
- Move Vale directory out from Netlify into its own directory
- Add a new Make target to run Vale on changed .md files prior to `git commit`
- Use new vocab structure in docs here: https://vale.sh/docs/topics/vocab/
- Update Vale action to use Reviewdog
 - Vale no longer fails when it finds an error in the text.

### CODE REVIEW

File: v22.2/add-column.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ docs_area: reference.sql
 
 {% include {{ page.version.version }}/sql/combine-alter-table-commands.md %}
 
+This is a new sentence.
+
+Bababa testtest
+test test test test test test this should fail repetition asdflkmakl;m4w3;lraksfmdl;xcvkn;4uitnu3warksfljdfndlasjkfnlakjwrnalfkjsndkljn
```
Comment (mdlinville) [inline v22.2/add-column.md:17]: This seems non-ideal to have a separate failure for each repetition.

File: vale/Vocab/CockroachDB/accept.txt [hunk 2]
```diff
@@ -23,17 +17,10 @@ autovacuum
 avro
 AZ
 AZs
```
Comment (mdlinville) [inline vale/Vocab/CockroachDB/accept.txt:19]: Could this and the preceding be combined into a regex?

**Review by mdlinville — COMMENTED** [review]
Does it make sense to add some reject rules?

File: v22.2/add-column.md (hunk 1 above)
Comment (nickvigilante) [reply v22.2/add-column.md:17]: I can possibly tweak the RegEx to catch one or more repetitions.

Comment (nickvigilante) [reply v22.2/add-column.md:17]: Found a better RegEx.

File: vale/Vocab/CockroachDB/accept.txt (hunk 2 above)
Comment (nickvigilante) [reply vale/Vocab/CockroachDB/accept.txt:19]: I was going to work on that in more detail in a separate PR, but I've fixed that line specifically for this PR.

**Comment by nickvigilante** [comment]
> Does it make sense to add some reject rules?

@mdlinville We can, but 1) I don't know any reject rules we need right now. Can you think of any? Also, would this block successful merging of this PR?

**Comment by mdlinville** [comment]
> > Does it make sense to add some reject rules?
> 
> @mdlinville We can, but 1) I don't know any reject rules we need right now. Can you think of any? Also, would this block successful merging of this PR?

No, I can't think of any that aren't already covered by another test. It makes sense to use an iterative approach.

**Review by mdlinville — APPROVED** [review]

**Pull request merged**